plugins/pci-psp/fu-plugin-pci-psp.c

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (C) 2022 Advanced Micro Devices Inc.
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+struct FuPluginData {
+	gchar *sysfs_path;
+};
+
+static void
+fu_plugin_pci_psp_init(FuPlugin *plugin)
+{
+	fu_plugin_alloc_data(plugin, sizeof(FuPluginData));
+	fu_plugin_add_udev_subsystem(plugin, "pci");
+}
+
+static void
+fu_plugin_pci_psp_destroy(FuPlugin *plugin)
+{
+	FuPluginData *priv = fu_plugin_get_data(plugin);
+	g_free(priv->sysfs_path);
+}
+
+static gboolean
+fu_plugin_pci_psp_backend_device_added(FuPlugin *plugin, FuDevice *device, GError **error)
+{
+	FuPluginData *priv = fu_plugin_get_data(plugin);
+
+	/* interesting device? */
+	if (!FU_IS_UDEV_DEVICE(device))
+		return TRUE;
+	if (g_strcmp0(fu_udev_device_get_subsystem(FU_UDEV_DEVICE(device)), "pci") != 0)
+		return TRUE;
+
+	priv->sysfs_path = g_strdup(fu_udev_device_get_sysfs_path(FU_UDEV_DEVICE(device)));
+
+	return TRUE;
+}
+
+static gboolean
+fu_plugin_pci_psp_get_attr(FuPlugin *plugin,
+			   FwupdSecurityAttr *attr,
+			   const gchar *file,
+			   gboolean *out,
+			   GError **error)
+{
+	FuPluginData *priv = fu_plugin_get_data(plugin);
+	g_autofree gchar *fn = g_build_filename(priv->sysfs_path, file, NULL);
+	g_autofree gchar *buf = NULL;
+	gsize bufsz = 0;
+
+	if (!g_file_get_contents(fn, &buf, &bufsz, error)) {
+		g_prefix_error(error, "could not open %s: ", fn);
+		fwupd_security_attr_add_flag(attr, FWUPD_SECURITY_ATTR_FLAG_MISSING_DATA);
+		return FALSE;
+	}
+	*out = fu_common_strtoull(buf) ? TRUE : FALSE;
+
+	return TRUE;
+}
+#define simple_security_attr(attr_name, level, name, sysfs, fail, success, OBS)                    \
+	static void fu_plugin_pci_psp_security_attr_##name(FuPlugin *plugin,                       \
+							   FuSecurityAttrs *attrs)                 \
```

Comment:
not really a fan of mega macros like this

Reply:
Yeah I left it split out because I wasn't sure either on it. I'll take it out.